File: week11/Counter.py
class Counter:
    def __init__(self, name):
        self.name = name
        self.count = 0
    
    def Increment(self):
        self.count += 1
    
    def Reset(self):
        self.count = 0
    
    @property
    def Name(self):
        return self.name
    
    @Name.setter
    def Name(self, value):
        self.name = value
    
    @property
    def Ticks(self):
        return self.count

File: week3/CounterTask2/CounterTask2/Counter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CounterTask2
{
    public class Counter(string name)
    {
        private int _count = 0;
        private string _name = name;

        public void Increment()
        {
            _count++;
        }

        public void Reset()
        {
            _count = 0;
        }

        public string Name
        {
            get
            {
                return _name;
            }
            set
            {
                _name = value;
            }
        }
        public int Tick
        {
            get
            {
                return _count;
            }
        }
    }
}

